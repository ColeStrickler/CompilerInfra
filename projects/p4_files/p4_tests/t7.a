swag: int;


swag2 : custom {
    a : int;
    b : bool;

    swagyolo : (arg1: int, b: bool, c : int) -> bool
    {
        a = c * arg1;
        return a == 4;
    }
};



swag3 : (arg1 : int, arg2 : bool) -> bool {
    arg1 = swag3(1, false);
    if (arg2)
    {
        cType : swag2;
        arg1 = cType->a;
        cType->swagyolo(arg1, arg2, 1);
    }
}