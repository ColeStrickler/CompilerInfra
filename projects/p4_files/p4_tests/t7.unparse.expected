swag{int} : int;
swag2{swag2} : custom {
    a{int} : int;
    b{bool} : bool;
    swagyolo{(int, bool, int)->bool} : (arg1{int} : int, b{bool} : bool, c{int} : int) -> bool {
        a{int} = c{int} * arg1{int};
        return a{int} == 4;
    }
};
swag3{(int, bool)->bool} : (arg1{int} : int, arg2{bool} : bool) -> bool {
    arg1{int} = swag3{(int, bool)->bool}(1, false);
    if (arg2{bool}){
        cType{swag2} : swag2{swag2};
        arg1{int} = cType{swag2}->a{int};
        cType{swag2}->swagyolo{(int, bool, int)->bool}(arg1{int}, arg2{bool}, 1);
    }
}